lib/peer.ml

```diff
@@ -1,26 +1,24 @@
-type state =
+type status =
   | Alive
   | Suspicious
   | Faulty
 
 type t = {
-  address : string;
-  port : int;
-  known_peers : t list;
-  state : state;
+  address : Address.t;
+  mutable status : status;
+  mutable peers : (Address.t, t) Base.Hashtbl.t;
 }
 
```

Comment:
I think this belongs in the failure detector internally. The peer type isn't really concerned with this type of hashtbl.

Reply:
Referring here to `retrieve_peer_from_address_opt`

Reply:
Actually, it is. Since a peer must knows a list of peers. I went for `Hashtbl` for convenience.